hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/MemStoreCompactor.java

```diff
@@ -254,6 +258,11 @@ private ImmutableSegment compact(int numOfCells) throws IOException {
             compactingMemStore.getConfiguration(), compactingMemStore.getComparator(), iterator,
             numOfCells, ImmutableSegment.Type.ARRAY_MAP_BASED);
         break;
+      case MERGE_TO_ARRAY_MAP:
```

Comment:
6 parameters ... very complex. Just shows why we need a small hierarchy of immutable segments.